demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -215,14 +362,423 @@ static bool getS3ObjectFileSize( size_t * pFileSize,
+        {
+            LogError( ( "Failed to add x-amz-iot-thing-name header to request headers: Error=%s.",
+                        HTTPClient_strerror( httpStatus ) ) );
+            returnStatus = false;
+        }
+    }
+
+    if( returnStatus == true )
+    {
+        /* Send the request and receive the response. */
+        httpStatus = HTTPClient_Send( transportInterface,
+                                      &requestHeaders,
+                                      NULL,
+                                      0,
+                                      response, 0 );
+
+        if( httpStatus != HTTPSuccess )
+        {
+            LogError( ( "Failed to send HTTP GET request to %s%s: Error=%s.",
+                        pAddress, pPath, HTTPClient_strerror( httpStatus ) ) );
+            returnStatus = false;
+        }
+    }
+
+    if( returnStatus == true )
+    {
+        /* Parse the credentials received in the response. */
+        jsonStatus = parseCredentials( response, sigvCreds );
+
+        if( jsonStatus != JSONSuccess )
+        {
+            LogError( ( "Failed to parse temporary IOT credentials. " ) );
+            returnStatus = false;
+        }
+    }
+
+    /* Get the AWS IOT date from the http response. */
+    if( returnStatus == true )
+    {
+        httpStatus = HTTPClient_ReadHeader( response,
+                                            HTTP_DEMO_RECEIVED_DATE_HEADER_FIELD,
+                                            sizeof( HTTP_DEMO_RECEIVED_DATE_HEADER_FIELD ) - 1,
+                                            ( const char ** ) &pDate,
+                                            &dateLen );
+
+        if( httpStatus != HTTPSuccess )
+        {
+            LogError( ( "Failed to retrieve %s header from response: Error=%s.",
+                        HTTP_DEMO_RECEIVED_DATE_HEADER_FIELD, HTTPClient_strerror( httpStatus ) ) );
+            returnStatus = false;
+        }
+    }
+
+    if( returnStatus == true )
+    {
+        /* Convert AWS IOT date retrieved from IOT server to ISO 8601 date format. */
+        sigv4Status = SigV4_AwsIotDateToIso8601( pDate, dateLen, pDateISO8601, pDateISO8601Len );
+
+        if( sigv4Status != SigV4Success )
+        {
+            LogError( ( "Failed to convert AWS IOT date to ISO 8601 format." ) );
+            returnStatus = false;
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static JSONStatus_t parseCredentials( HTTPResponse_t * response,
+                                      SigV4Credentials_t * sigvCreds )
+{
+    JSONStatus_t jsonStatus = JSONSuccess;
+
+    assert( response != NULL );
+    assert( sigvCreds != NULL );
+
+    if( jsonStatus == JSONSuccess )
+    {
+        /* Get accessKeyId from HTTP response. */
+        jsonStatus = JSON_Search( ( char * ) response->pBody,
+                                  response->bodyLen,
+                                  "credentials.accessKeyId",
```

Comment:
It is important to store these string literals of JSON keys in macros for easier visibility and readability.
```
#define CREDENTIALS_RESPONSE_ACCESS_KEY_ID_KEY "credentials.accessKeyId"
```

Reply:
Similary for `"credentials.secretAccessKey"` and `"credentials.sessionToken"`